src/peergos/server/corenode/UsernameValidator.java

```diff
@@ -9,17 +11,18 @@
  */
 public final class UsernameValidator {
 
-    final static Pattern VALID_USERNAME = Pattern.compile("^(?=.{1,32}$)(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+(?<![_.])$");
+    private static final Pattern VALID_USERNAME = Pattern.compile("^(?=.{1,32}$)(?![_-])(?!.*[_-]{2})[a-z0-9_-]+(?<![_-])$");
 
     /** Username rules:
-     * no _ or . at the end
-     * allowed characters [a-zA-Z0-9._]
-     * no __ or _. or ._ or .. inside
-     * no _ or . at the beginning
+     * no _- at the end
+     * allowed characters [a-z0-9_-]
+     * no __ or -- or _- or -_ inside
+     * no _- at the beginning
```

Comment:
So ``.`` is allowed at the start of a username? Just checking.

Reply:
. isn't allowed anywhere